jax/_src/logging_config.py

```diff
@@ -13,19 +13,100 @@
 # limitations under the License.
 
 import logging
+import os
 import sys
 
-_debug_handler = logging.StreamHandler(sys.stderr)
-_debug_handler.setLevel(logging.DEBUG)
 # Example log message:
 # DEBUG:2023-06-07 00:14:40,280:jax._src.xla_bridge:590: Initializing backend 'cpu'
-_debug_handler.setFormatter(logging.Formatter(
-    "{levelname}:{asctime}:{name}:{lineno}: {message}", style='{'))
+logging_formatter = logging.Formatter(
+    "{levelname}:{asctime}:{name}:{lineno}: {message}", style='{')
+
+_logging_level_handler_set: dict[str, tuple[logging.Handler, int]] = {}
+_default_TF_CPP_MIN_LOG_LEVEL = os.environ.get("TF_CPP_MIN_LOG_LEVEL", "1")
+
+_nameToLevel = {
+    'CRITICAL': logging.CRITICAL,
+    'FATAL': logging.FATAL,
+    'ERROR': logging.ERROR,
+    'WARN': logging.WARNING,
+    'WARNING': logging.WARNING,
+    'INFO': logging.INFO,
+    'DEBUG': logging.DEBUG,
+    'NOTSET': logging.NOTSET,
+}
+def _getLevelNamesMapping():
+  return _nameToLevel
+
+
+def _logging_level_to_int(logging_level: str):
+  # attempt to convert the logging level to integer
+  try:
+    # logging level is a string representation of an integer
+    logging_level_num = int(logging_level)
+  except ValueError:
+    # logging level is a name string
+    logging_level_num = _getLevelNamesMapping()[logging_level]
+  return logging_level_num
+
+_tf_cpp_map = {
+    'CRITICAL': 3,
+    'FATAL': 3,
+    'ERROR': 2,
+    'WARN': 1,
+    'WARNING': 1,
+    'INFO': 0,
+    'DEBUG': 0,
+}
+
+def _set_TF_CPP_MIN_LOG_LEVEL(logging_level: str | None = None):
+  # resetting to user-default TF_CPP_MIN_LOG_LEVEL
+  # this is typically "1", but if the user overrode it, it can be != "1"
+  os.environ["TF_CPP_MIN_LOG_LEVEL"] = _default_TF_CPP_MIN_LOG_LEVEL
+
+  # set cpp runtime logging level if the level is anything but NOTSET
+  if logging_level is not None and logging_level != "NOTSET":
+    if logging_level not in _tf_cpp_map:
+      raise ValueError(f"Attempting to set log level \"{logging_level}\" which"
+                       f" isn't one of the supported:"
+                       f" {list(_tf_cpp_map.keys())}.")
+    # config the CPP logging level 0 - debug, 1 - info, 2 - warning, 3 - error
+    os.environ["TF_CPP_MIN_LOG_LEVEL"] = str(_tf_cpp_map[logging_level])
+
+def update_logging_level_global(logging_level: str | None) -> None:
```

Comment:
This doesn't appear to interact at all with the debug logging module right? How do you adding two debug handlers?

Now that I think of it, maybe it's fine to just have multiple log output...